system/jlib/jdebug.cpp

```diff
@@ -1475,9 +1475,8 @@ class CProcessMonitor
             processes.item(i1).active = false;
         DIR *dir = opendir("/proc");
         loop {
-            struct dirent *ent;
-            struct dirent entryMem;
-            readdir_r(dir, &entryMem, &ent);
+            CriticalBlock b(sect);
+            struct dirent *ent = readdir(dir);
```

Comment:
I believe this CriticalBlock will stay in effect for the entirety of each dirent use, and be reset on each loop iteration.  I don't normally get to play with our threading library though so please double check.  @mckellyln